### ISSUE TITLE
The SSH transport process has abruptly terminated causing this remote session to break (write - ERROR:109 on prior unblocking write)

### ISSUE THREAD

**Issue by StoneFlower70** [PowerShell/Win32-OpenSSH#1755]
**Troubleshooting steps**
https://github.com/PowerShell/Win32-OpenSSH/wiki/Troubleshooting-Steps

**Terminal issue? please go through wiki**
https://github.com/PowerShell/Win32-OpenSSH/wiki/TTY-PTY-support-in-Windows-OpenSSH

Please answer the following

**"OpenSSH for Windows" version**
7.1.3
 
**Server OperatingSystem**
Windows Server 2019 Standard

**Client OperatingSystem**
CentOS Linux release 7.9.2009 (Core)

**What is failing**
Enter-PSSession : The SSH client session has ended with error message: The SSH transport process has abruptly terminated causing this remote session to break.

**Expected output**
Connect to Windows Server via SSH

**Actual output**
Enter-PSSession: The SSH client session has ended with error message: The SSH transport process has abruptly terminated causing this remote session to break.

Here is the sshd.log from the server (Windows) box. Debug3 set.

7644 2021-03-19 15:41:36.307 debug2: fd 3 setting O_NONBLOCK
7644 2021-03-19 15:41:36.307 debug3: sock_set_v6only: set socket 3 IPV6_V6ONLY
7644 2021-03-19 15:41:36.307 debug1: Bind to port 22 on ::.
7644 2021-03-19 15:41:36.307 Server listening on :: port 22.
7644 2021-03-19 15:41:36.307 debug2: fd 4 setting O_NONBLOCK
7644 2021-03-19 15:41:36.307 debug1: Bind to port 22 on 0.0.0.0.
7644 2021-03-19 15:41:36.307 Server listening on 0.0.0.0 port 22.
7644 2021-03-19 15:41:42.867 debug3: fd 5 is not O_NONBLOCK
7644 2021-03-19 15:41:42.868 debug3: spawning "C:\\WINDOWS\\System32\\OpenSSH\\sshd.exe" "-R"
7644 2021-03-19 15:41:42.869 debug3: send_rexec_state: entering fd = 8 config len 284
7644 2021-03-19 15:41:42.871 debug3: ssh_msg_send: type 0
7644 2021-03-19 15:41:42.871 debug3: send_rexec_state: done
12228 2021-03-19 15:41:42.903 debug1: inetd sockets after dupping: 3, 3
12228 2021-03-19 15:41:42.904 Connection from 10.200.20.17 port 57762 on 10.200.20.150 port 22
12228 2021-03-19 15:41:42.904 debug1: Client protocol version 2.0; client software version OpenSSH_7.4
12228 2021-03-19 15:41:42.904 debug1: match: OpenSSH_7.4 pat OpenSSH* compat 0x04000000
12228 2021-03-19 15:41:42.904 debug1: Local version string SSH-2.0-OpenSSH_for_Windows_7.7
12228 2021-03-19 15:41:42.904 debug2: fd 3 setting O_NONBLOCK
12228 2021-03-19 15:41:42.976 debug3: spawning "C:\\WINDOWS\\System32\\OpenSSH\\sshd.exe" "-y"
12228 2021-03-19 15:41:42.978 debug2: Network child is on pid 11208
12228 2021-03-19 15:41:42.978 debug3: send_rexec_state: entering fd = 6 config len 284
12228 2021-03-19 15:41:42.978 debug3: ssh_msg_send: type 0
12228 2021-03-19 15:41:42.978 debug3: send_rexec_state: done
12228 2021-03-19 15:41:42.978 debug3: ssh_msg_send: type 0
12228 2021-03-19 15:41:42.978 debug3: ssh_msg_send: type 0
12228 2021-03-19 15:41:42.978 debug3: preauth child monitor started
12228 2021-03-19 15:41:42.991 debug1: list_hostkey_types: ssh-rsa,rsa-sha2-512,rsa-sha2-256,ecdsa-sha2-nistp256,ssh-ed25519 [preauth]
12228 2021-03-19 15:41:42.991 debug3: send packet: type 20 [preauth]
12228 2021-03-19 15:41:42.991 debug1: SSH2_MSG_KEXINIT sent [preauth]
12228 2021-03-19 15:41:42.991 debug3: receive packet: type 20 [preauth]
12228 2021-03-19 15:41:42.991 debug1: SSH2_MSG_KEXINIT received [preauth]
12228 2021-03-19 15:41:42.991 debug2: local server KEXINIT proposal [preauth]
12228 2021-03-19 15:41:42.991 debug2: KEX algorithms: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1 [preauth]
12228 2021-03-19 15:41:42.991 debug2: host key algorithms: ssh-rsa,rsa-sha2-512,rsa-sha2-256,ecdsa-sha2-nistp256,ssh-ed25519 [preauth]
12228 2021-03-19 15:41:42.991 debug2: ciphers ctos: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com [preauth]
12228 2021-03-19 15:41:42.991 debug2: ciphers stoc: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com [preauth]
12228 2021-03-19 15:41:42.991 debug2: MACs ctos: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1 [preauth]
12228 2021-03-19 15:41:42.991 debug2: MACs stoc: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1 [preauth]
12228 2021-03-19 15:41:42.991 debug2: compression ctos: none [preauth]
12228 2021-03-19 15:41:42.991 debug2: compression stoc: none [preauth]
12228 2021-03-19 15:41:42.991 debug2: languages ctos:  [preauth]
12228 2021-03-19 15:41:42.991 debug2: languages stoc:  [preauth]
12228 2021-03-19 15:41:42.991 debug2: first_kex_follows 0  [preauth]
12228 2021-03-19 15:41:42.991 debug2: reserved 0  [preauth]
12228 2021-03-19 15:41:42.991 debug2: peer client KEXINIT proposal [preauth]
12228 2021-03-19 15:41:42.991 debug2: KEX algorithms: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1,diffie-hellman-group1-sha1,ext-info-c [preauth]
12228 2021-03-19 15:41:42.991 debug2: host key algorithms: ecdsa-sha2-nistp256-cert-v01@openssh.com,ecdsa-sha2-nistp384-cert-v01@openssh.com,ecdsa-sha2-nistp521-cert-v01@openssh.com,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-ed25519-cert-v01@openssh.com,ssh-rsa-cert-v01@openssh.com,ssh-dss-cert-v01@openssh.com,ssh-ed25519,rsa-sha2-512,rsa-sha2-256,ssh-rsa,ssh-dss [preauth]
12228 2021-03-19 15:41:42.991 debug2: ciphers ctos: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com,aes128-cbc,aes192-cbc,aes256-cbc [preauth]
12228 2021-03-19 15:41:42.991 debug2: ciphers stoc: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com,aes128-cbc,aes192-cbc,aes256-cbc [preauth]
12228 2021-03-19 15:41:42.991 debug2: MACs ctos: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1 [preauth]
12228 2021-03-19 15:41:42.991 debug2: MACs stoc: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1 [preauth]
12228 2021-03-19 15:41:42.991 debug2: compression ctos: none,zlib@openssh.com,zlib [preauth]
12228 2021-03-19 15:41:42.991 debug2: compression stoc: none,zlib@openssh.com,zlib [preauth]
12228 2021-03-19 15:41:42.991 debug2: languages ctos:  [preauth]
12228 2021-03-19 15:41:42.991 debug2: languages stoc:  [preauth]
12228 2021-03-19 15:41:42.991 debug2: first_kex_follows 0  [preauth]
12228 2021-03-19 15:41:42.991 debug2: reserved 0  [preauth]
12228 2021-03-19 15:41:42.991 debug1: kex: algorithm: curve25519-sha256 [preauth]
12228 2021-03-19 15:41:42.991 debug1: kex: host key algorithm: ecdsa-sha2-nistp256 [preauth]
12228 2021-03-19 15:41:42.991 debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none [preauth]
12228 2021-03-19 15:41:42.991 debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none [preauth]
12228 2021-03-19 15:41:42.991 debug1: expecting SSH2_MSG_KEX_ECDH_INIT [preauth]
12228 2021-03-19 15:41:42.993 debug3: receive packet: type 30 [preauth]
12228 2021-03-19 15:41:42.996 debug3: mm_key_sign entering [preauth]
12228 2021-03-19 15:41:42.996 debug3: mm_request_send entering: type 6 [preauth]
12228 2021-03-19 15:41:42.996 debug3: mm_request_receive entering
12228 2021-03-19 15:41:42.996 debug3: monitor_read: checking request 6
12228 2021-03-19 15:41:42.996 debug3: mm_answer_sign
12228 2021-03-19 15:41:42.997 debug3: mm_answer_sign: hostkey proof signature 000002428302E9D0(100)
12228 2021-03-19 15:41:42.997 debug3: mm_request_send entering: type 7
12228 2021-03-19 15:41:42.997 debug2: monitor_read: 6 used once, disabling now
12228 2021-03-19 15:41:42.997 debug3: mm_key_sign: waiting for MONITOR_ANS_SIGN [preauth]
12228 2021-03-19 15:41:42.997 debug3: mm_request_receive_expect entering: type 7 [preauth]
12228 2021-03-19 15:41:42.997 debug3: mm_request_receive entering [preauth]
12228 2021-03-19 15:41:42.997 debug3: send packet: type 31 [preauth]
12228 2021-03-19 15:41:42.997 debug3: send packet: type 21 [preauth]
12228 2021-03-19 15:41:42.997 debug2: set_newkeys: mode 1 [preauth]
12228 2021-03-19 15:41:42.997 debug1: rekey after 134217728 blocks [preauth]
12228 2021-03-19 15:41:42.997 debug1: SSH2_MSG_NEWKEYS sent [preauth]
12228 2021-03-19 15:41:42.997 debug1: expecting SSH2_MSG_NEWKEYS [preauth]
12228 2021-03-19 15:41:42.997 debug3: send packet: type 7 [preauth]
12228 2021-03-19 15:41:43.000 debug3: receive packet: type 21 [preauth]
12228 2021-03-19 15:41:43.000 debug1: SSH2_MSG_NEWKEYS received [preauth]
12228 2021-03-19 15:41:43.000 debug2: set_newkeys: mode 0 [preauth]
12228 2021-03-19 15:41:43.000 debug1: rekey after 134217728 blocks [preauth]
12228 2021-03-19 15:41:43.000 debug1: KEX done [preauth]
12228 2021-03-19 15:41:43.015 debug3: receive packet: type 5 [preauth]
12228 2021-03-19 15:41:43.015 debug3: send packet: type 6 [preauth]
12228 2021-03-19 15:41:43.015 debug3: receive packet: type 50 [preauth]
12228 2021-03-19 15:41:43.015 debug1: userauth-request for user [redacted]@[redacted] service ssh-connection method none [preauth]
12228 2021-03-19 15:41:43.015 debug1: attempt 0 failures 0 [preauth]
12228 2021-03-19 15:41:43.015 debug3: mm_getpwnamallow entering [preauth]
12228 2021-03-19 15:41:43.015 debug3: mm_request_send entering: type 8 [preauth]
12228 2021-03-19 15:41:43.015 debug3: mm_getpwnamallow: waiting for MONITOR_ANS_PWNAM [preauth]
12228 2021-03-19 15:41:43.015 debug3: mm_request_receive_expect entering: type 9 [preauth]
12228 2021-03-19 15:41:43.015 debug3: mm_request_receive entering [preauth]
12228 2021-03-19 15:41:43.015 debug3: mm_request_receive entering
12228 2021-03-19 15:41:43.015 debug3: monitor_read: checking request 8
12228 2021-03-19 15:41:43.015 debug3: mm_answer_pwnamallow
12228 2021-03-19 15:41:43.016 debug2: parse_server_config: config reprocess config len 284
12228 2021-03-19 15:41:43.016 debug3: mm_answer_pwnamallow: sending MONITOR_ANS_PWNAM: 1
12228 2021-03-19 15:41:43.016 debug3: mm_request_send entering: type 9
12228 2021-03-19 15:41:43.016 debug2: monitor_read: 8 used once, disabling now
12228 2021-03-19 15:41:43.016 debug2: input_userauth_request: setting up authctxt for [redacted]@[redacted] [preauth]
12228 2021-03-19 15:41:43.016 debug3: mm_inform_authserv entering [preauth]
12228 2021-03-19 15:41:43.016 debug3: mm_request_send entering: type 4 [preauth]
12228 2021-03-19 15:41:43.016 debug2: input_userauth_request: try method none [preauth]
12228 2021-03-19 15:41:43.016 debug3: userauth_finish: failure partial=0 next methods="publickey,password,keyboard-interactive" [preauth]
12228 2021-03-19 15:41:43.016 debug3: send packet: type 51 [preauth]
12228 2021-03-19 15:41:43.016 debug3: mm_request_receive entering
12228 2021-03-19 15:41:43.016 debug3: monitor_read: checking request 4
12228 2021-03-19 15:41:43.016 debug3: mm_answer_authserv: service=ssh-connection, style=
12228 2021-03-19 15:41:43.016 debug2: monitor_read: 4 used once, disabling now
12228 2021-03-19 15:41:43.016 debug3: receive packet: type 50 [preauth]
12228 2021-03-19 15:41:43.016 debug1: userauth-request for user [redacted]@[redacted] service ssh-connection method publickey [preauth]
12228 2021-03-19 15:41:43.016 debug1: attempt 1 failures 0 [preauth]
12228 2021-03-19 15:41:43.016 debug2: input_userauth_request: try method publickey [preauth]
12228 2021-03-19 15:41:43.016 debug1: userauth_pubkey: test pkalg rsa-sha2-512 pkblob RSA SHA256:[redacted] [preauth]
12228 2021-03-19 15:41:43.016 debug3: mm_key_allowed entering [preauth]
12228 2021-03-19 15:41:43.016 debug3: mm_request_send entering: type 22 [preauth]
12228 2021-03-19 15:41:43.016 debug3: mm_key_allowed: waiting for MONITOR_ANS_KEYALLOWED [preauth]
12228 2021-03-19 15:41:43.016 debug3: mm_request_receive_expect entering: type 23 [preauth]
12228 2021-03-19 15:41:43.016 debug3: mm_request_receive entering [preauth]
12228 2021-03-19 15:41:43.016 debug3: mm_request_receive entering
12228 2021-03-19 15:41:43.016 debug3: monitor_read: checking request 22
12228 2021-03-19 15:41:43.016 debug3: mm_answer_keyallowed entering
12228 2021-03-19 15:41:43.016 debug3: mm_answer_keyallowed: key_from_blob: 000002428302EDC0
12228 2021-03-19 15:41:43.016 debug1: trying public key file C:\\Users\\[redacted]\\.ssh/authorized_keys
12228 2021-03-19 15:41:43.017 debug1: C:\\Users\\[redacted]\\.ssh/authorized_keys:2: matching key found: RSA SHA256:[redacted]
12228 2021-03-19 15:41:43.017 debug1: C:\\Users\\[redacted]\\.ssh/authorized_keys:2: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
12228 2021-03-19 15:41:43.017 Accepted key RSA SHA256:[redacted] found at C:\\Users\\[redacted]\\.ssh/authorized_keys:2
12228 2021-03-19 15:41:43.017 debug3: mm_answer_keyallowed: publickey authentication test: RSA key is allowed
12228 2021-03-19 15:41:43.017 debug3: mm_request_send entering: type 23
12228 2021-03-19 15:41:43.017 debug3: send packet: type 60 [preauth]
12228 2021-03-19 15:41:43.017 debug2: userauth_pubkey: authenticated 0 pkalg rsa-sha2-512 [preauth]
12228 2021-03-19 15:41:43.017 Postponed publickey for [redacted]@[redacted] from 10.200.20.17 port 57762 ssh2 [preauth]
12228 2021-03-19 15:41:43.018 debug3: receive packet: type 50 [preauth]
12228 2021-03-19 15:41:43.019 debug1: userauth-request for user [redacted]@[redacted] service ssh-connection method publickey [preauth]
12228 2021-03-19 15:41:43.019 debug1: attempt 2 failures 0 [preauth]
12228 2021-03-19 15:41:43.019 debug2: input_userauth_request: try method publickey [preauth]
12228 2021-03-19 15:41:43.019 debug3: userauth_pubkey: have rsa-sha2-512 signature for RSA SHA256:[redacted] [preauth]
12228 2021-03-19 15:41:43.019 debug3: mm_key_allowed entering [preauth]
12228 2021-03-19 15:41:43.019 debug3: mm_request_send entering: type 22 [preauth]
12228 2021-03-19 15:41:43.019 debug3: mm_key_allowed: waiting for MONITOR_ANS_KEYALLOWED [preauth]
12228 2021-03-19 15:41:43.019 debug3: mm_request_receive_expect entering: type 23 [preauth]
12228 2021-03-19 15:41:43.019 debug3: mm_request_receive entering [preauth]
12228 2021-03-19 15:41:43.019 debug3: mm_request_receive entering
12228 2021-03-19 15:41:43.019 debug3: monitor_read: checking request 22
12228 2021-03-19 15:41:43.019 debug3: mm_answer_keyallowed entering
12228 2021-03-19 15:41:43.019 debug3: mm_answer_keyallowed: key_from_blob: 000002428302E810
12228 2021-03-19 15:41:43.019 debug1: trying public key file C:\\Users\\[redacted]\\.ssh/authorized_keys
12228 2021-03-19 15:41:43.019 debug1: C:\\Users\\[redacted]\\.ssh/authorized_keys:2: matching key found: RSA SHA256:[redacted]
12228 2021-03-19 15:41:43.019 debug1: C:\\Users\\[redacted]\\.ssh/authorized_keys:2: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
12228 2021-03-19 15:41:43.019 Accepted key RSA SHA256:[redacted] found at C:\\Users\\[redacted]\\.ssh/authorized_keys:2
12228 2021-03-19 15:41:43.019 debug3: mm_answer_keyallowed: publickey authentication: RSA key is allowed
12228 2021-03-19 15:41:43.019 debug3: mm_request_send entering: type 23
12228 2021-03-19 15:41:43.019 debug3: mm_sshkey_verify entering [preauth]
12228 2021-03-19 15:41:43.019 debug3: mm_request_send entering: type 24 [preauth]
12228 2021-03-19 15:41:43.019 debug3: mm_sshkey_verify: waiting for MONITOR_ANS_KEYVERIFY [preauth]
12228 2021-03-19 15:41:43.019 debug3: mm_request_receive_expect entering: type 25 [preauth]
12228 2021-03-19 15:41:43.019 debug3: mm_request_receive entering [preauth]
12228 2021-03-19 15:41:43.019 debug3: mm_request_receive entering
12228 2021-03-19 15:41:43.019 debug3: monitor_read: checking request 24
12228 2021-03-19 15:41:43.020 debug3: mm_answer_keyverify: publickey 000002428302E9D0 signature verified
12228 2021-03-19 15:41:43.020 debug1: auth_activate_options: setting new authentication options
12228 2021-03-19 15:41:43.020 debug3: mm_request_send entering: type 25
12228 2021-03-19 15:41:43.020 Accepted publickey for [redacted]@[redacted] from 10.200.20.17 port 57762 ssh2: RSA SHA256:[redacted]
12228 2021-03-19 15:41:43.020 debug1: monitor_child_preauth: [redacted]@[redacted] has been authenticated by privileged process
12228 2021-03-19 15:41:43.020 debug3: mm_get_keystate: Waiting for new keys
12228 2021-03-19 15:41:43.020 debug3: mm_request_receive_expect entering: type 26
12228 2021-03-19 15:41:43.020 debug3: mm_request_receive entering
12228 2021-03-19 15:41:43.020 debug3: mm_get_keystate: GOT new keys
12228 2021-03-19 15:41:43.020 debug1: auth_activate_options: setting new authentication options [preauth]
12228 2021-03-19 15:41:43.020 debug2: userauth_pubkey: authenticated 1 pkalg rsa-sha2-512 [preauth]
12228 2021-03-19 15:41:43.020 debug3: send packet: type 52 [preauth]
12228 2021-03-19 15:41:43.020 debug3: mm_request_send entering: type 26 [preauth]
12228 2021-03-19 15:41:43.020 debug3: mm_send_keystate: Finished sending state [preauth]
12228 2021-03-19 15:41:43.021 debug1: monitor_read_log: child log fd closed
12228 2021-03-19 15:41:43.032 debug3: lookup_principal_name: Successfully discovered explicit principal name: '[redacted]\\[redacted]'=>'[redacted]@[redacted].com'
12228 2021-03-19 15:41:43.033 debug3: LsaLogonUser Succeeded (Impersonation: 1)
12228 2021-03-19 15:41:43.050 debug3: spawning "C:\\WINDOWS\\System32\\OpenSSH\\sshd.exe" "-z"
12228 2021-03-19 15:41:43.051 User child is on pid 6540
12228 2021-03-19 15:41:43.051 debug3: send_rexec_state: entering fd = 5 config len 284
12228 2021-03-19 15:41:43.051 debug3: ssh_msg_send: type 0
12228 2021-03-19 15:41:43.051 debug3: send_rexec_state: done
12228 2021-03-19 15:41:43.051 debug3: ssh_msg_send: type 0
12228 2021-03-19 15:41:43.051 debug3: ssh_msg_send: type 0
12228 2021-03-19 15:41:43.051 debug3: ssh_msg_send: type 0
12228 2021-03-19 15:41:43.051 debug3: ssh_msg_send: type 0
12228 2021-03-19 15:41:43.051 debug3: ssh_msg_send: type 0
6540 2021-03-19 15:41:43.061 debug3: ssh_msg_recv entering
6540 2021-03-19 15:41:43.061 debug2: fd 4 setting O_NONBLOCK
6540 2021-03-19 15:41:43.061 debug3: ssh_msg_recv entering
6540 2021-03-19 15:41:43.062 debug2: parse_server_config: config reprocess config len 284
6540 2021-03-19 15:41:43.062 debug3: ssh_msg_recv entering
6540 2021-03-19 15:41:43.062 debug3: monitor_apply_keystate: packet_set_state
6540 2021-03-19 15:41:43.062 debug2: set_newkeys: mode 0
6540 2021-03-19 15:41:43.062 debug1: rekey after 134217728 blocks
6540 2021-03-19 15:41:43.062 debug2: set_newkeys: mode 1
6540 2021-03-19 15:41:43.062 debug1: rekey after 134217728 blocks
6540 2021-03-19 15:41:43.062 debug1: ssh_packet_set_postauth: called
6540 2021-03-19 15:41:43.062 debug3: ssh_packet_set_state: done
6540 2021-03-19 15:41:43.062 debug3: ssh_msg_recv entering
6540 2021-03-19 15:41:43.062 debug3: notify_hostkeys: key 0: ssh-rsa SHA256:[redacted]
6540 2021-03-19 15:41:43.062 debug3: notify_hostkeys: key 1: ecdsa-sha2-nistp256 SHA256:[redacted]
6540 2021-03-19 15:41:43.062 debug3: notify_hostkeys: key 2: ssh-ed25519 SHA256:[redacted]
6540 2021-03-19 15:41:43.062 debug3: notify_hostkeys: sent 3 hostkeys
6540 2021-03-19 15:41:43.062 debug3: send packet: type 80
6540 2021-03-19 15:41:43.063 debug1: active: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
6540 2021-03-19 15:41:43.063 debug1: Entering interactive session for SSH2.
6540 2021-03-19 15:41:43.063 debug2: fd 7 setting O_NONBLOCK
6540 2021-03-19 15:41:43.063 debug2: fd 8 setting O_NONBLOCK
6540 2021-03-19 15:41:43.063 debug1: server_init_dispatch
6540 2021-03-19 15:41:43.063 debug3: receive packet: type 90
6540 2021-03-19 15:41:43.063 debug1: server_input_channel_open: ctype session rchan 0 win 2097152 max 32768
6540 2021-03-19 15:41:43.063 debug1: input_session_request
6540 2021-03-19 15:41:43.063 debug1: channel 0: new [server-session]
6540 2021-03-19 15:41:43.063 debug2: session_new: allocate (allocated 0 max 10)
6540 2021-03-19 15:41:43.063 debug3: session_unused: session id 0 unused
6540 2021-03-19 15:41:43.063 debug1: session_new: session 0
6540 2021-03-19 15:41:43.063 debug1: session_open: channel 0
6540 2021-03-19 15:41:43.063 debug1: session_open: session 0: link with channel 0
6540 2021-03-19 15:41:43.063 debug1: server_input_channel_open: confirm session
6540 2021-03-19 15:41:43.063 debug3: send packet: type 91
6540 2021-03-19 15:41:43.063 debug3: receive packet: type 80
6540 2021-03-19 15:41:43.063 debug1: server_input_global_request: rtype no-more-sessions@openssh.com want_reply 0
6540 2021-03-19 15:41:43.083 debug3: receive packet: type 98
6540 2021-03-19 15:41:43.083 debug1: server_input_channel_req: channel 0 request env reply 0
6540 2021-03-19 15:41:43.083 debug1: session_by_channel: session 0 channel 0
6540 2021-03-19 15:41:43.083 debug1: session_input_channel_req: session 0 req env
6540 2021-03-19 15:41:43.083 debug2: Ignoring env request LANG: disallowed name
6540 2021-03-19 15:41:43.083 debug3: receive packet: type 98
6540 2021-03-19 15:41:43.083 debug1: server_input_channel_req: channel 0 request subsystem reply 1
6540 2021-03-19 15:41:43.083 debug1: session_by_channel: session 0 channel 0
6540 2021-03-19 15:41:43.083 debug1: session_input_channel_req: session 0 req subsystem
6540 2021-03-19 15:41:43.083 debug2: subsystem request for powershell by user [redacted]\\[redacted]
6540 2021-03-19 15:41:43.083 debug1: subsystem: exec() C:/Program Files/PowerShell/7/pwsh.exe -sshs -NoLogo -NoProfile
6540 2021-03-19 15:41:43.083 Starting session: subsystem 'powershell' for [redacted]\\[redacted] from 10.200.20.17 port 57762 id 0
6540 2021-03-19 15:41:43.083 debug2: fd 9 setting O_NONBLOCK
6540 2021-03-19 15:41:43.083 debug2: fd 10 setting O_NONBLOCK
6540 2021-03-19 15:41:43.083 debug2: fd 11 setting O_NONBLOCK
6540 2021-03-19 15:41:43.083 debug2: fd 12 setting O_NONBLOCK
6540 2021-03-19 15:41:43.083 debug2: fd 13 setting O_NONBLOCK
6540 2021-03-19 15:41:43.083 debug2: fd 14 setting O_NONBLOCK
6540 2021-03-19 15:41:43.083 debug1: Executing command: "c:\\windows\\system32\\cmd.exe" /c "C:/Program Files/PowerShell/7/pwsh.exe -sshs -NoLogo -NoProfile" with no pty
6540 2021-03-19 15:41:43.086 debug2: fd 4 setting TCP_NODELAY
6540 2021-03-19 15:41:43.086 debug3: fd 11 is O_NONBLOCK
6540 2021-03-19 15:41:43.086 debug3: fd 10 is O_NONBLOCK
6540 2021-03-19 15:41:43.086 debug3: fd 13 is O_NONBLOCK
6540 2021-03-19 15:41:43.086 debug3: send packet: type 99
6540 2021-03-19 15:41:43.091 debug2: channel 0: read 101 from efd 13
6540 2021-03-19 15:41:43.091 debug3: channel 0: discard efd
6540 2021-03-19 15:41:43.107 debug2: channel 0: read<=0 rfd 11 len 0
6540 2021-03-19 15:41:43.107 debug2: channel 0: read failed
6540 2021-03-19 15:41:43.107 debug2: channel 0: close_read
6540 2021-03-19 15:41:43.107 debug2: channel 0: input open -> drain
6540 2021-03-19 15:41:43.107 debug3: write - ERROR:109 on prior unblocking write, io:00000273578CE450
6540 2021-03-19 15:41:43.107 debug2: channel 0: write failed
6540 2021-03-19 15:41:43.107 debug2: channel 0: close_write
6540 2021-03-19 15:41:43.107 debug2: channel 0: send eow
6540 2021-03-19 15:41:43.107 debug2: channel 0: output open -> closed
6540 2021-03-19 15:41:43.107 debug2: channel 0: read 0 from efd 13
6540 2021-03-19 15:41:43.107 debug2: channel 0: closing read-efd 13
6540 2021-03-19 15:41:43.107 debug2: channel 0: ibuf empty
6540 2021-03-19 15:41:43.107 debug2: channel 0: send eof
6540 2021-03-19 15:41:43.107 debug3: send packet: type 96
6540 2021-03-19 15:41:43.107 debug2: channel 0: input drain -> closed
6540 2021-03-19 15:41:43.108 debug1: Received SIGCHLD.
6540 2021-03-19 15:41:43.108 debug1: session_by_pid: pid 12560
6540 2021-03-19 15:41:43.108 debug1: session_exit_message: session 0 channel 0 pid 12560
6540 2021-03-19 15:41:43.108 debug2: channel 0: request exit-status confirm 0
6540 2021-03-19 15:41:43.108 debug3: send packet: type 98
6540 2021-03-19 15:41:43.108 debug1: session_exit_message: release channel 0
6540 2021-03-19 15:41:43.108 debug2: channel 0: send close
6540 2021-03-19 15:41:43.108 debug3: send packet: type 97
6540 2021-03-19 15:41:43.108 debug3: channel 0: will not send data after close
6540 2021-03-19 15:41:43.108 debug2: notify_done: reading
6540 2021-03-19 15:41:43.108 debug3: channel 0: will not send data after close
6540 2021-03-19 15:41:43.108 debug3: receive packet: type 97
6540 2021-03-19 15:41:43.108 debug2: channel 0: rcvd close
6540 2021-03-19 15:41:43.108 debug3: channel 0: will not send data after close
6540 2021-03-19 15:41:43.108 debug2: channel 0: is dead
6540 2021-03-19 15:41:43.108 debug2: channel 0: gc: notify user
6540 2021-03-19 15:41:43.108 debug1: session_by_channel: session 0 channel 0
6540 2021-03-19 15:41:43.108 debug1: session_close_by_channel: channel 0 child 0
6540 2021-03-19 15:41:43.108 Close session: user [redacted]\\[redacted] from 10.200.20.17 port 57762 id 0
6540 2021-03-19 15:41:43.108 debug3: session_unused: session id 0 unused
6540 2021-03-19 15:41:43.108 debug2: channel 0: gc: user detached
6540 2021-03-19 15:41:43.108 debug2: channel 0: is dead
6540 2021-03-19 15:41:43.108 debug2: channel 0: garbage collecting
6540 2021-03-19 15:41:43.108 debug1: channel 0: free: server-session, nchannels 1
6540 2021-03-19 15:41:43.108 debug3: channel 0: status: The following connections are open:\r\n  #0 server-session (t4 r0 i3/0 o3/0 fd -1/-1 cc -1)\r\n
6540 2021-03-19 15:41:43.112 debug3: receive packet: type 1
6540 2021-03-19 15:41:43.112 Received disconnect from 10.200.20.17 port 57762:11: disconnected by user
6540 2021-03-19 15:41:43.112 Disconnected from 10.200.20.17 port 57762
6540 2021-03-19 15:41:43.112 debug1: do_cleanup
12228 2021-03-19 15:41:43.113 debug3: mm_request_receive entering
12228 2021-03-19 15:41:43.113 debug1: do_cleanup

**Comment by StoneFlower70** [comment]
Also, I have reviewed #1259 from @mikebuckley and I have confirmed that the pwsh.exe is being properly instantiated on my Windows server.

**Comment by bagajjal** [comment]
@StoneFlower70 - You are using very old version. Try with [OpenSSH v8.1](https://github.com/PowerShell/Win32-OpenSSH/releases/tag/v8.1.0.0p1-Beta)

**Comment by StoneFlower70** [comment]
@bagajjal I may have trouble getting a beta version approved for Prod. But it's worth a try. Thank you for the suggestion.

**Comment by bagajjal** [comment]
@StoneFlower70 - I would first repro the problem on a non-prod machine then upgrade to v8.1 to see if it resolves your issue.

**Comment by StoneFlower70** [comment]
@bagajjal - I have installed v8.1 on a Windows 10 Pro x64 box. Unfortunately, I am seeing the same error as before:
debug3: write - ERROR:109 on prior unblocking write, io:0000026CB86D3670

Here is the full log:
29852 2021-03-22 16:50:43.685 debug2: fd 3 setting O_NONBLOCK
29852 2021-03-22 16:50:43.685 debug3: sock_set_v6only: set socket 3 IPV6_V6ONLY
29852 2021-03-22 16:50:43.685 debug1: Bind to port 22 on ::.
29852 2021-03-22 16:50:43.686 Server listening on :: port 22.
29852 2021-03-22 16:50:43.686 debug2: fd 4 setting O_NONBLOCK
29852 2021-03-22 16:50:43.686 debug1: Bind to port 22 on 0.0.0.0.
29852 2021-03-22 16:50:43.686 Server listening on 0.0.0.0 port 22.
29852 2021-03-22 16:50:57.267 debug3: fd 5 is not O_NONBLOCK
29852 2021-03-22 16:50:57.268 debug3: spawning "C:\\Users\\[redacted]\\Downloads\\OpenSSH-Win64-v8.1.0.0p1-Beta\\OpenSSH-Win64\\sshd.exe" -R
29852 2021-03-22 16:50:57.270 debug3: send_rexec_state: entering fd = 8 config len 381
29852 2021-03-22 16:50:57.272 debug3: ssh_msg_send: type 0
29852 2021-03-22 16:50:57.272 debug3: send_rexec_state: done
39252 2021-03-22 16:50:57.288 debug1: inetd sockets after dupping: 4, 4
39252 2021-03-22 16:50:57.288 Connection from 10.200.20.17 port 46926 on 10.20.2.56 port 22
39252 2021-03-22 16:50:57.288 debug1: Local version string SSH-2.0-OpenSSH_for_Windows_8.1
39252 2021-03-22 16:50:57.288 debug1: Remote protocol version 2.0, remote software version OpenSSH_7.4
39252 2021-03-22 16:50:57.288 debug1: match: OpenSSH_7.4 pat OpenSSH_7.0*,OpenSSH_7.1*,OpenSSH_7.2*,OpenSSH_7.3*,OpenSSH_7.4*,OpenSSH_7.5*,OpenSSH_7.6*,OpenSSH_7.7* compat 0x04000002
39252 2021-03-22 16:50:57.288 debug2: fd 4 setting O_NONBLOCK
39252 2021-03-22 16:50:57.324 debug3: spawning "C:\\Users\\[redacted]\\Downloads\\OpenSSH-Win64-v8.1.0.0p1-Beta\\OpenSSH-Win64\\sshd.exe" -y
39252 2021-03-22 16:50:57.327 debug2: Network child is on pid 21540
39252 2021-03-22 16:50:57.327 debug3: send_rexec_state: entering fd = 6 config len 381
39252 2021-03-22 16:50:57.327 debug3: ssh_msg_send: type 0
39252 2021-03-22 16:50:57.327 debug3: send_rexec_state: done
39252 2021-03-22 16:50:57.327 debug3: ssh_msg_send: type 0
39252 2021-03-22 16:50:57.327 debug3: ssh_msg_send: type 0
39252 2021-03-22 16:50:57.327 debug3: preauth child monitor started
39252 2021-03-22 16:50:57.340 debug1: list_hostkey_types: rsa-sha2-512,rsa-sha2-256,ssh-rsa,ecdsa-sha2-nistp256,ssh-ed25519 [preauth]
39252 2021-03-22 16:50:57.340 debug3: send packet: type 20 [preauth]
39252 2021-03-22 16:50:57.340 debug1: SSH2_MSG_KEXINIT sent [preauth]
39252 2021-03-22 16:50:57.340 debug3: receive packet: type 20 [preauth]
39252 2021-03-22 16:50:57.340 debug1: SSH2_MSG_KEXINIT received [preauth]
39252 2021-03-22 16:50:57.340 debug2: local server KEXINIT proposal [preauth]
39252 2021-03-22 16:50:57.340 debug2: KEX algorithms: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1 [preauth]
39252 2021-03-22 16:50:57.340 debug2: host key algorithms: rsa-sha2-512,rsa-sha2-256,ssh-rsa,ecdsa-sha2-nistp256,ssh-ed25519 [preauth]
39252 2021-03-22 16:50:57.340 debug2: ciphers ctos: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com [preauth]
39252 2021-03-22 16:50:57.340 debug2: ciphers stoc: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com [preauth]
39252 2021-03-22 16:50:57.340 debug2: MACs ctos: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1 [preauth]
39252 2021-03-22 16:50:57.340 debug2: MACs stoc: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1 [preauth]
39252 2021-03-22 16:50:57.340 debug2: compression ctos: none,zlib@openssh.com [preauth]
39252 2021-03-22 16:50:57.340 debug2: compression stoc: none,zlib@openssh.com [preauth]
39252 2021-03-22 16:50:57.340 debug2: languages ctos:  [preauth]
39252 2021-03-22 16:50:57.340 debug2: languages stoc:  [preauth]
39252 2021-03-22 16:50:57.340 debug2: first_kex_follows 0  [preauth]
39252 2021-03-22 16:50:57.340 debug2: reserved 0  [preauth]
39252 2021-03-22 16:50:57.340 debug2: peer client KEXINIT proposal [preauth]
39252 2021-03-22 16:50:57.340 debug2: KEX algorithms: curve25519-sha256,curve25519-sha256@libssh.org,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521,diffie-hellman-group-exchange-sha256,diffie-hellman-group16-sha512,diffie-hellman-group18-sha512,diffie-hellman-group-exchange-sha1,diffie-hellman-group14-sha256,diffie-hellman-group14-sha1,diffie-hellman-group1-sha1,ext-info-c [preauth]
39252 2021-03-22 16:50:57.340 debug2: host key algorithms: ecdsa-sha2-nistp256-cert-v01@openssh.com,ecdsa-sha2-nistp384-cert-v01@openssh.com,ecdsa-sha2-nistp521-cert-v01@openssh.com,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-ed25519-cert-v01@openssh.com,ssh-rsa-cert-v01@openssh.com,ssh-dss-cert-v01@openssh.com,ssh-ed25519,rsa-sha2-512,rsa-sha2-256,ssh-rsa,ssh-dss [preauth]
39252 2021-03-22 16:50:57.340 debug2: ciphers ctos: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com,aes128-cbc,aes192-cbc,aes256-cbc [preauth]
39252 2021-03-22 16:50:57.340 debug2: ciphers stoc: chacha20-poly1305@openssh.com,aes128-ctr,aes192-ctr,aes256-ctr,aes128-gcm@openssh.com,aes256-gcm@openssh.com,aes128-cbc,aes192-cbc,aes256-cbc [preauth]
39252 2021-03-22 16:50:57.340 debug2: MACs ctos: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1 [preauth]
39252 2021-03-22 16:50:57.340 debug2: MACs stoc: umac-64-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-256-etm@openssh.com,hmac-sha2-512-etm@openssh.com,hmac-sha1-etm@openssh.com,umac-64@openssh.com,umac-128@openssh.com,hmac-sha2-256,hmac-sha2-512,hmac-sha1 [preauth]
39252 2021-03-22 16:50:57.340 debug2: compression ctos: none,zlib@openssh.com,zlib [preauth]
39252 2021-03-22 16:50:57.341 debug2: compression stoc: none,zlib@openssh.com,zlib [preauth]
39252 2021-03-22 16:50:57.341 debug2: languages ctos:  [preauth]
39252 2021-03-22 16:50:57.341 debug2: languages stoc:  [preauth]
39252 2021-03-22 16:50:57.341 debug2: first_kex_follows 0  [preauth]
39252 2021-03-22 16:50:57.341 debug2: reserved 0  [preauth]
39252 2021-03-22 16:50:57.341 debug1: kex: algorithm: curve25519-sha256 [preauth]
39252 2021-03-22 16:50:57.341 debug1: kex: host key algorithm: ecdsa-sha2-nistp256 [preauth]
39252 2021-03-22 16:50:57.341 debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none [preauth]
39252 2021-03-22 16:50:57.341 debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none [preauth]
39252 2021-03-22 16:50:57.341 debug1: expecting SSH2_MSG_KEX_ECDH_INIT [preauth]
39252 2021-03-22 16:50:57.349 debug3: receive packet: type 30 [preauth]
39252 2021-03-22 16:50:57.351 debug3: mm_sshkey_sign entering [preauth]
39252 2021-03-22 16:50:57.351 debug3: mm_request_send entering: type 6 [preauth]
39252 2021-03-22 16:50:57.351 debug3: mm_sshkey_sign: waiting for MONITOR_ANS_SIGN [preauth]
39252 2021-03-22 16:50:57.351 debug3: mm_request_receive_expect entering: type 7 [preauth]
39252 2021-03-22 16:50:57.351 debug3: mm_request_receive entering [preauth]
39252 2021-03-22 16:50:57.351 debug3: mm_request_receive entering
39252 2021-03-22 16:50:57.351 debug3: monitor_read: checking request 6
39252 2021-03-22 16:50:57.351 debug3: mm_answer_sign
39252 2021-03-22 16:50:57.353 debug3: mm_answer_sign: KEX signature 0000024BE0A849F0(100)
39252 2021-03-22 16:50:57.353 debug3: mm_request_send entering: type 7
39252 2021-03-22 16:50:57.353 debug2: monitor_read: 6 used once, disabling now
39252 2021-03-22 16:50:57.354 debug3: send packet: type 31 [preauth]
39252 2021-03-22 16:50:57.354 debug3: send packet: type 21 [preauth]
39252 2021-03-22 16:50:57.354 debug2: set_newkeys: mode 1 [preauth]
39252 2021-03-22 16:50:57.354 debug1: rekey out after 134217728 blocks [preauth]
39252 2021-03-22 16:50:57.354 debug1: SSH2_MSG_NEWKEYS sent [preauth]
39252 2021-03-22 16:50:57.354 debug1: Sending SSH2_MSG_EXT_INFO [preauth]
39252 2021-03-22 16:50:57.354 debug3: send packet: type 7 [preauth]
39252 2021-03-22 16:50:57.354 debug1: expecting SSH2_MSG_NEWKEYS [preauth]
39252 2021-03-22 16:50:57.363 debug3: receive packet: type 21 [preauth]
39252 2021-03-22 16:50:57.363 debug1: SSH2_MSG_NEWKEYS received [preauth]
39252 2021-03-22 16:50:57.363 debug2: set_newkeys: mode 0 [preauth]
39252 2021-03-22 16:50:57.363 debug1: rekey in after 134217728 blocks [preauth]
39252 2021-03-22 16:50:57.363 debug1: KEX done [preauth]
39252 2021-03-22 16:50:57.410 debug3: receive packet: type 5 [preauth]
39252 2021-03-22 16:50:57.410 debug3: send packet: type 6 [preauth]
39252 2021-03-22 16:50:57.415 debug3: receive packet: type 50 [preauth]
39252 2021-03-22 16:50:57.415 debug1: userauth-request for user [redacted]@[redacted] service ssh-connection method none [preauth]
39252 2021-03-22 16:50:57.415 debug1: attempt 0 failures 0 [preauth]
39252 2021-03-22 16:50:57.415 debug3: mm_getpwnamallow entering [preauth]
39252 2021-03-22 16:50:57.415 debug3: mm_request_send entering: type 8 [preauth]
39252 2021-03-22 16:50:57.415 debug3: mm_getpwnamallow: waiting for MONITOR_ANS_PWNAM [preauth]
39252 2021-03-22 16:50:57.415 debug3: mm_request_receive_expect entering: type 9 [preauth]
39252 2021-03-22 16:50:57.415 debug3: mm_request_receive entering [preauth]
39252 2021-03-22 16:50:57.415 debug3: mm_request_receive entering
39252 2021-03-22 16:50:57.415 debug3: monitor_read: checking request 8
39252 2021-03-22 16:50:57.415 debug3: mm_answer_pwnamallow
39252 2021-03-22 16:50:57.418 debug2: parse_server_config: config reprocess config len 381
39252 2021-03-22 16:50:57.418 debug3: checking match for 'Group administrators' user [redacted]\\[redacted] host 10.200.20.17 addr 10.200.20.17 laddr 10.20.2.56 lport 22
39252 2021-03-22 16:50:57.443 debug3: lookup_principal_name: Successfully discovered explicit principal name: '[redacted]\\[redacted]'=>'[redacted]@[redacted].com'
39252 2021-03-22 16:50:57.456 debug3: LsaLogonUser Succeeded (Impersonation: 0)
39252 2021-03-22 16:50:57.456 debug1: user [redacted]\\[redacted] matched group list administrators at line 93
39252 2021-03-22 16:50:57.456 debug3: match found
39252 2021-03-22 16:50:57.456 debug3: reprocess config:94 setting AuthorizedKeysFile __PROGRAMDATA__/ssh/administrators_authorized_keys
39252 2021-03-22 16:50:57.457 debug3: mm_answer_pwnamallow: sending MONITOR_ANS_PWNAM: 1
39252 2021-03-22 16:50:57.457 debug3: mm_request_send entering: type 9
39252 2021-03-22 16:50:57.457 debug2: monitor_read: 8 used once, disabling now
39252 2021-03-22 16:50:57.458 debug2: input_userauth_request: setting up authctxt for [redacted]@[redacted] [preauth]
39252 2021-03-22 16:50:57.458 debug3: mm_inform_authserv entering [preauth]
39252 2021-03-22 16:50:57.458 debug3: mm_request_send entering: type 4 [preauth]
39252 2021-03-22 16:50:57.458 debug2: input_userauth_request: try method none [preauth]
39252 2021-03-22 16:50:57.458 debug3: user_specific_delay: user specific delay 0.000ms [preauth]
39252 2021-03-22 16:50:57.458 debug3: ensure_minimum_time_since: elapsed 43.761ms, delaying 37.985ms (requested 5.109ms) [preauth]
39252 2021-03-22 16:50:57.458 debug3: mm_request_receive entering
39252 2021-03-22 16:50:57.458 debug3: monitor_read: checking request 4
39252 2021-03-22 16:50:57.458 debug3: mm_answer_authserv: service=ssh-connection, style=
39252 2021-03-22 16:50:57.458 debug2: monitor_read: 4 used once, disabling now
39252 2021-03-22 16:50:57.497 debug3: userauth_finish: failure partial=0 next methods="publickey,password,keyboard-interactive" [preauth]
39252 2021-03-22 16:50:57.497 debug3: send packet: type 51 [preauth]
39252 2021-03-22 16:50:57.503 debug3: receive packet: type 50 [preauth]
39252 2021-03-22 16:50:57.503 debug1: userauth-request for user [redacted]@[redacted] service ssh-connection method publickey [preauth]
39252 2021-03-22 16:50:57.503 debug1: attempt 1 failures 0 [preauth]
39252 2021-03-22 16:50:57.503 debug2: input_userauth_request: try method publickey [preauth]
39252 2021-03-22 16:50:57.503 debug2: userauth_pubkey: valid user [redacted]@[redacted] querying public key rsa-sha2-512 [redacted] [preauth]
39252 2021-03-22 16:50:57.503 debug1: userauth_pubkey: test pkalg rsa-sha2-512 pkblob RSA SHA256:[redacted] [preauth]
39252 2021-03-22 16:50:57.503 debug3: mm_key_allowed entering [preauth]
39252 2021-03-22 16:50:57.503 debug3: mm_request_send entering: type 22 [preauth]
39252 2021-03-22 16:50:57.503 debug3: mm_key_allowed: waiting for MONITOR_ANS_KEYALLOWED [preauth]
39252 2021-03-22 16:50:57.503 debug3: mm_request_receive_expect entering: type 23 [preauth]
39252 2021-03-22 16:50:57.503 debug3: mm_request_receive entering [preauth]
39252 2021-03-22 16:50:57.503 debug3: mm_request_receive entering
39252 2021-03-22 16:50:57.503 debug3: monitor_read: checking request 22
39252 2021-03-22 16:50:57.503 debug3: mm_answer_keyallowed entering
39252 2021-03-22 16:50:57.503 debug3: mm_answer_keyallowed: key_from_blob: 0000024BE0ABB1F0
39252 2021-03-22 16:50:57.503 debug1: trying public key file __PROGRAMDATA__/ssh/administrators_authorized_keys
39252 2021-03-22 16:50:57.505 debug1: __PROGRAMDATA__/ssh/administrators_authorized_keys:1: matching key found: RSA SHA256:[redacted]
39252 2021-03-22 16:50:57.505 debug1: __PROGRAMDATA__/ssh/administrators_authorized_keys:1: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
39252 2021-03-22 16:50:57.505 Accepted key RSA SHA256:[redacted] found at __PROGRAMDATA__/ssh/administrators_authorized_keys:1
39252 2021-03-22 16:50:57.505 debug3: mm_answer_keyallowed: publickey authentication test: RSA key is allowed
39252 2021-03-22 16:50:57.505 debug3: mm_request_send entering: type 23
39252 2021-03-22 16:50:57.505 debug3: send packet: type 60 [preauth]
39252 2021-03-22 16:50:57.505 debug2: userauth_pubkey: authenticated 0 pkalg rsa-sha2-512 [preauth]
39252 2021-03-22 16:50:57.505 debug3: user_specific_delay: user specific delay 0.000ms [preauth]
39252 2021-03-22 16:50:57.505 debug3: ensure_minimum_time_since: elapsed 1.950ms, delaying 3.159ms (requested 5.109ms) [preauth]
39252 2021-03-22 16:50:57.509 Postponed publickey for [redacted]@[redacted] from 10.200.20.17 port 46926 ssh2 [preauth]
39252 2021-03-22 16:50:57.512 debug3: receive packet: type 50 [preauth]
39252 2021-03-22 16:50:57.512 debug1: userauth-request for user [redacted]@[redacted] service ssh-connection method publickey [preauth]
39252 2021-03-22 16:50:57.512 debug1: attempt 2 failures 0 [preauth]
39252 2021-03-22 16:50:57.512 debug2: input_userauth_request: try method publickey [preauth]
39252 2021-03-22 16:50:57.512 debug2: userauth_pubkey: valid user [redacted]@[redacted] attempting public key rsa-sha2-512 [redacted] [preauth]
39252 2021-03-22 16:50:57.512 debug3: userauth_pubkey: have rsa-sha2-512 signature for RSA SHA256:[redacted] [preauth]
39252 2021-03-22 16:50:57.512 debug3: mm_key_allowed entering [preauth]
39252 2021-03-22 16:50:57.512 debug3: mm_request_send entering: type 22 [preauth]
39252 2021-03-22 16:50:57.512 debug3: mm_key_allowed: waiting for MONITOR_ANS_KEYALLOWED [preauth]
39252 2021-03-22 16:50:57.512 debug3: mm_request_receive_expect entering: type 23 [preauth]
39252 2021-03-22 16:50:57.512 debug3: mm_request_receive entering [preauth]
39252 2021-03-22 16:50:57.512 debug3: mm_request_receive entering
39252 2021-03-22 16:50:57.512 debug3: monitor_read: checking request 22
39252 2021-03-22 16:50:57.512 debug3: mm_answer_keyallowed entering
39252 2021-03-22 16:50:57.512 debug3: mm_answer_keyallowed: key_from_blob: 0000024BE0ABB1F0
39252 2021-03-22 16:50:57.513 debug1: trying public key file __PROGRAMDATA__/ssh/administrators_authorized_keys
39252 2021-03-22 16:50:57.514 debug1: __PROGRAMDATA__/ssh/administrators_authorized_keys:1: matching key found: RSA [redacted]
39252 2021-03-22 16:50:57.514 debug1: __PROGRAMDATA__/ssh/administrators_authorized_keys:1: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
39252 2021-03-22 16:50:57.514 Accepted key RSA SHA256:[redacted] found at __PROGRAMDATA__/ssh/administrators_authorized_keys:1
39252 2021-03-22 16:50:57.514 debug3: mm_answer_keyallowed: publickey authentication: RSA key is allowed
39252 2021-03-22 16:50:57.514 debug3: mm_request_send entering: type 23
39252 2021-03-22 16:50:57.514 debug3: mm_sshkey_verify entering [preauth]
39252 2021-03-22 16:50:57.514 debug3: mm_request_send entering: type 24 [preauth]
39252 2021-03-22 16:50:57.515 debug3: mm_sshkey_verify: waiting for MONITOR_ANS_KEYVERIFY [preauth]
39252 2021-03-22 16:50:57.515 debug3: mm_request_receive_expect entering: type 25 [preauth]
39252 2021-03-22 16:50:57.515 debug3: mm_request_receive entering [preauth]
39252 2021-03-22 16:50:57.515 debug3: mm_request_receive entering
39252 2021-03-22 16:50:57.515 debug3: monitor_read: checking request 24
39252 2021-03-22 16:50:57.515 debug3: mm_answer_keyverify: publickey 0000024BE0ABA740 signature verified
39252 2021-03-22 16:50:57.515 debug1: auth_activate_options: setting new authentication options
39252 2021-03-22 16:50:57.515 debug3: mm_request_send entering: type 25
39252 2021-03-22 16:50:57.515 Accepted publickey for [redacted]@[redacted] from 10.200.20.17 port 46926 ssh2: RSA SHA256:[redacted]
39252 2021-03-22 16:50:57.515 debug1: monitor_child_preauth: [redacted]@[redacted] has been authenticated by privileged process
39252 2021-03-22 16:50:57.515 debug3: mm_get_keystate: Waiting for new keys
39252 2021-03-22 16:50:57.515 debug3: mm_request_receive_expect entering: type 26
39252 2021-03-22 16:50:57.515 debug3: mm_request_receive entering
39252 2021-03-22 16:50:57.518 debug3: mm_get_keystate: GOT new keys
39252 2021-03-22 16:50:57.518 debug1: auth_activate_options: setting new authentication options [preauth]
39252 2021-03-22 16:50:57.518 debug2: userauth_pubkey: authenticated 1 pkalg rsa-sha2-512 [preauth]
39252 2021-03-22 16:50:57.518 debug3: user_specific_delay: user specific delay 0.000ms [preauth]
39252 2021-03-22 16:50:57.518 debug3: ensure_minimum_time_since: elapsed 3.852ms, delaying 1.257ms (requested 5.109ms) [preauth]
39252 2021-03-22 16:50:57.518 debug3: send packet: type 52 [preauth]
39252 2021-03-22 16:50:57.518 debug3: mm_request_send entering: type 26 [preauth]
39252 2021-03-22 16:50:57.518 debug3: mm_send_keystate: Finished sending state [preauth]
39252 2021-03-22 16:50:57.520 debug1: monitor_read_log: child log fd closed
39252 2021-03-22 16:50:57.530 debug3: lookup_principal_name: Successfully discovered explicit principal name: '[redacted]\\[redacted]'=>'[redacted]@[redacted].com'
39252 2021-03-22 16:50:57.537 debug3: LsaLogonUser Succeeded (Impersonation: 1)
39252 2021-03-22 16:50:57.548 debug3: spawning "C:\\Users\\[redacted]\\Downloads\\OpenSSH-Win64-v8.1.0.0p1-Beta\\OpenSSH-Win64\\sshd.exe" -z
39252 2021-03-22 16:50:57.553 User child is on pid 39372
39252 2021-03-22 16:50:57.553 debug3: send_rexec_state: entering fd = 5 config len 381
39252 2021-03-22 16:50:57.553 debug3: ssh_msg_send: type 0
39252 2021-03-22 16:50:57.553 debug3: send_rexec_state: done
39252 2021-03-22 16:50:57.553 debug3: ssh_msg_send: type 0
39252 2021-03-22 16:50:57.553 debug3: ssh_msg_send: type 0
39252 2021-03-22 16:50:57.553 debug3: ssh_msg_send: type 0
39252 2021-03-22 16:50:57.553 debug3: ssh_msg_send: type 0
39252 2021-03-22 16:50:57.553 debug3: ssh_msg_send: type 0
39372 2021-03-22 16:50:57.566 debug3: recv_idexch_state: entering fd = 3
39372 2021-03-22 16:50:57.566 debug3: ssh_msg_recv entering
39372 2021-03-22 16:50:57.566 debug3: recv_idexch_state: done
39372 2021-03-22 16:50:57.566 debug2: fd 4 setting O_NONBLOCK
39372 2021-03-22 16:50:57.566 debug3: recv_autxctx_state: entering fd = 3
39372 2021-03-22 16:50:57.566 debug3: ssh_msg_recv entering
39372 2021-03-22 16:50:57.566 debug3: recv_autxctx_state: done
39372 2021-03-22 16:50:57.567 debug2: parse_server_config: config reprocess config len 381
39372 2021-03-22 16:50:57.567 debug3: checking match for 'Group administrators' user [redacted]\\[redacted] host 10.200.20.17 addr 10.200.20.17 laddr 10.20.2.56 lport 22
39372 2021-03-22 16:50:57.567 debug3: get_user_token - i am running as [redacted]\\[redacted], returning process token
39372 2021-03-22 16:50:57.567 debug1: user [redacted]\\[redacted] matched group list administrators at line 93
39372 2021-03-22 16:50:57.567 debug3: match found
39372 2021-03-22 16:50:57.567 debug3: reprocess config:94 setting AuthorizedKeysFile __PROGRAMDATA__/ssh/administrators_authorized_keys
39372 2021-03-22 16:50:57.568 debug3: monitor_recv_keystate: entering 
39372 2021-03-22 16:50:57.568 debug3: ssh_msg_recv entering
39372 2021-03-22 16:50:57.568 debug3: monitor_recv_keystate: done
39372 2021-03-22 16:50:57.568 debug3: monitor_apply_keystate: packet_set_state
39372 2021-03-22 16:50:57.568 debug2: set_newkeys: mode 0
39372 2021-03-22 16:50:57.568 debug1: rekey in after 134217728 blocks
39372 2021-03-22 16:50:57.568 debug2: set_newkeys: mode 1
39372 2021-03-22 16:50:57.568 debug1: rekey out after 134217728 blocks
39372 2021-03-22 16:50:57.568 debug1: ssh_packet_set_postauth: called
39372 2021-03-22 16:50:57.568 debug3: ssh_packet_set_state: done
39372 2021-03-22 16:50:57.568 debug3: monitor_recv_authopt: entering 
39372 2021-03-22 16:50:57.568 debug3: ssh_msg_recv entering
39372 2021-03-22 16:50:57.568 debug3: monitor_recv_authopt: done
39372 2021-03-22 16:50:57.568 debug3: notify_hostkeys: key 0: ssh-rsa SHA256:[redacted]
39372 2021-03-22 16:50:57.568 debug3: notify_hostkeys: key 1: ecdsa-sha2-nistp256 SHA256:[redacted]
39372 2021-03-22 16:50:57.568 debug3: notify_hostkeys: key 2: ssh-ed25519 SHA256:[redacted]
39372 2021-03-22 16:50:57.568 debug3: notify_hostkeys: sent 3 hostkeys
39372 2021-03-22 16:50:57.568 debug3: send packet: type 80
39372 2021-03-22 16:50:57.568 debug1: active: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding
39372 2021-03-22 16:50:57.568 debug1: Entering interactive session for SSH2.
39372 2021-03-22 16:50:57.568 debug2: fd 7 setting O_NONBLOCK
39372 2021-03-22 16:50:57.568 debug2: fd 8 setting O_NONBLOCK
39372 2021-03-22 16:50:57.568 debug1: server_init_dispatch
39372 2021-03-22 16:50:57.568 debug3: receive packet: type 90
39372 2021-03-22 16:50:57.568 debug1: server_input_channel_open: ctype session rchan 0 win 2097152 max 32768
39372 2021-03-22 16:50:57.568 debug1: input_session_request
39372 2021-03-22 16:50:57.568 debug1: channel 0: new [server-session]
39372 2021-03-22 16:50:57.568 debug2: session_new: allocate (allocated 0 max 10)
39372 2021-03-22 16:50:57.568 debug3: session_unused: session id 0 unused
39372 2021-03-22 16:50:57.568 debug1: session_new: session 0
39372 2021-03-22 16:50:57.568 debug1: session_open: channel 0
39372 2021-03-22 16:50:57.568 debug1: session_open: session 0: link with channel 0
39372 2021-03-22 16:50:57.568 debug1: server_input_channel_open: confirm session
39372 2021-03-22 16:50:57.568 debug3: send packet: type 91
39372 2021-03-22 16:50:57.568 debug3: receive packet: type 80
39372 2021-03-22 16:50:57.568 debug1: server_input_global_request: rtype no-more-sessions@openssh.com want_reply 0
39372 2021-03-22 16:50:57.619 debug3: receive packet: type 98
39372 2021-03-22 16:50:57.619 debug1: server_input_channel_req: channel 0 request env reply 0
39372 2021-03-22 16:50:57.619 debug1: session_by_channel: session 0 channel 0
39372 2021-03-22 16:50:57.619 debug1: session_input_channel_req: session 0 req env
39372 2021-03-22 16:50:57.619 debug2: Ignoring env request LANG: disallowed name
39372 2021-03-22 16:50:57.619 debug3: receive packet: type 98
39372 2021-03-22 16:50:57.619 debug1: server_input_channel_req: channel 0 request subsystem reply 1
39372 2021-03-22 16:50:57.619 debug1: session_by_channel: session 0 channel 0
39372 2021-03-22 16:50:57.619 debug1: session_input_channel_req: session 0 req subsystem
39372 2021-03-22 16:50:57.619 debug2: subsystem request for powershell by user [redacted]\\[redacted]
39372 2021-03-22 16:50:57.619 debug1: subsystem: exec() C:\\Program Files\\PowerShell\\7\\pwsh.exe -sshs -NoLogo -NoProfile
39372 2021-03-22 16:50:57.619 Starting session: subsystem 'powershell' for [redacted]\\[redacted] from 10.200.20.17 port 46926 id 0
39372 2021-03-22 16:50:57.619 debug2: fd 9 setting O_NONBLOCK
39372 2021-03-22 16:50:57.619 debug2: fd 10 setting O_NONBLOCK
39372 2021-03-22 16:50:57.619 debug2: fd 11 setting O_NONBLOCK
39372 2021-03-22 16:50:57.619 debug2: fd 12 setting O_NONBLOCK
39372 2021-03-22 16:50:57.619 debug2: fd 13 setting O_NONBLOCK
39372 2021-03-22 16:50:57.619 debug2: fd 14 setting O_NONBLOCK
39372 2021-03-22 16:50:57.619 debug3: shell: "c:\\windows\\system32\\cmd.exe"
39372 2021-03-22 16:50:57.619 debug3: shell_option: /c
39372 2021-03-22 16:50:57.619 debug3: exec_command: C:\\Program Files\\PowerShell\\7\\pwsh.exe -sshs -NoLogo -NoProfile
39372 2021-03-22 16:50:57.619 debug3: arg escape option: TRUE
39372 2021-03-22 16:50:57.619 debug3: spawn_argv[0]: "c:\\windows\\system32\\cmd.exe" /c "C:\\Program Files\\PowerShell\\7\\pwsh.exe -sshs -NoLogo -NoProfile"
39372 2021-03-22 16:50:57.619 debug3: spawning "c:\\windows\\system32\\cmd.exe" /c "C:\\Program Files\\PowerShell\\7\\pwsh.exe -sshs -NoLogo -NoProfile"
39372 2021-03-22 16:50:57.624 debug2: fd 4 setting TCP_NODELAY
39372 2021-03-22 16:50:57.624 debug3: fd 11 is O_NONBLOCK
39372 2021-03-22 16:50:57.624 debug3: fd 10 is O_NONBLOCK
39372 2021-03-22 16:50:57.624 debug3: fd 13 is O_NONBLOCK
39372 2021-03-22 16:50:57.624 debug3: send packet: type 99
39372 2021-03-22 16:50:57.651 debug2: channel 0: read 101 from efd 13
39372 2021-03-22 16:50:57.651 debug3: channel 0: discard efd
39372 2021-03-22 16:50:57.652 debug2: channel 0: read<=0 rfd 11 len 0
39372 2021-03-22 16:50:57.652 debug2: channel 0: read failed
39372 2021-03-22 16:50:57.652 debug2: channel 0: chan_shutdown_read (i0 o0 sock -1 wfd 11 efd 13 [ignore])
39372 2021-03-22 16:50:57.652 debug2: channel 0: input open -> drain
39372 2021-03-22 16:50:57.652 debug3: write - ERROR:109 on prior unblocking write, io:0000026CB86D3670
39372 2021-03-22 16:50:57.652 debug2: channel 0: write failed
39372 2021-03-22 16:50:57.652 debug2: channel 0: chan_shutdown_write (i1 o0 sock -1 wfd 10 efd 13 [ignore])
39372 2021-03-22 16:50:57.652 debug2: channel 0: send eow
39372 2021-03-22 16:50:57.652 debug2: channel 0: output open -> closed
39372 2021-03-22 16:50:57.652 debug2: channel 0: ibuf empty
39372 2021-03-22 16:50:57.652 debug2: channel 0: send eof
39372 2021-03-22 16:50:57.652 debug3: send packet: type 96
39372 2021-03-22 16:50:57.652 debug2: channel 0: input drain -> closed
39372 2021-03-22 16:50:57.653 debug1: Received SIGCHLD.
39372 2021-03-22 16:50:57.653 debug1: session_by_pid: pid 10476
39372 2021-03-22 16:50:57.653 debug1: session_exit_message: session 0 channel 0 pid 10476
39372 2021-03-22 16:50:57.653 debug2: channel 0: request exit-status confirm 0
39372 2021-03-22 16:50:57.653 debug3: send packet: type 98
39372 2021-03-22 16:50:57.653 debug1: session_exit_message: release channel 0
39372 2021-03-22 16:50:57.653 debug2: channel 0: send close
39372 2021-03-22 16:50:57.653 debug3: send packet: type 97
39372 2021-03-22 16:50:57.653 debug3: channel 0: will not send data after close
39372 2021-03-22 16:50:57.653 debug2: notify_done: reading
39372 2021-03-22 16:50:57.653 debug3: channel 0: will not send data after close
39372 2021-03-22 16:50:57.658 debug3: receive packet: type 97
39372 2021-03-22 16:50:57.658 debug2: channel 0: rcvd close
39372 2021-03-22 16:50:57.658 debug3: channel 0: will not send data after close
39372 2021-03-22 16:50:57.658 debug2: channel 0: is dead
39372 2021-03-22 16:50:57.658 debug2: channel 0: gc: notify user
39372 2021-03-22 16:50:57.658 debug1: session_by_channel: session 0 channel 0
39372 2021-03-22 16:50:57.658 debug1: session_close_by_channel: channel 0 child 0
39372 2021-03-22 16:50:57.658 Close session: user [redacted]\\[redacted] from 10.200.20.17 port 46926 id 0
39372 2021-03-22 16:50:57.658 debug3: session_unused: session id 0 unused
39372 2021-03-22 16:50:57.658 debug2: channel 0: gc: user detached
39372 2021-03-22 16:50:57.658 debug2: channel 0: is dead
39372 2021-03-22 16:50:57.658 debug2: channel 0: garbage collecting
39372 2021-03-22 16:50:57.658 debug1: channel 0: free: server-session, nchannels 1
39372 2021-03-22 16:50:57.658 debug3: channel 0: status: The following connections are open:\r\n  #0 server-session (t4 r0 i3/0 o3/0 e[ignore]/0 fd -1/-1/13 sock -1 cc -1)\r\n
39372 2021-03-22 16:50:57.658 debug3: receive packet: type 1
39372 2021-03-22 16:50:57.658 Received disconnect from 10.200.20.17 port 46926:11: disconnected by user
39372 2021-03-22 16:50:57.658 Disconnected from 10.200.20.17 port 46926
39372 2021-03-22 16:50:57.658 debug1: do_cleanup
39252 2021-03-22 16:50:57.660 debug3: mm_request_receive entering
39252 2021-03-22 16:50:57.660 debug1: do_cleanup

**Comment by bagajjal** [comment]
@StoneFlower70  - The problem is with the space.
The process creation is failing..
39372 2021-03-22 16:50:57.619 debug3: spawning "c:\windows\system32\cmd.exe" /c "C:\Program Files\PowerShell\7\pwsh.exe -sshs -NoLogo -NoProfile"  

Follow these steps,
1) Modify your sshd_config to use ""C:\Progra~1\PowerShell\7\pwsh.exe -sshs -NoLogo -NoProfile" instead of "C:\Program Files\PowerShell\7\pwsh.exe -sshs -NoLogo -NoProfile". 
2) Restart sshd service 
net stop sshd 
net start sshd
3) login now.
ssh user@ip

refer to https://github.com/PowerShell/Win32-OpenSSH/issues/784

**Comment by StoneFlower70** [comment]
Thank you @bagajjal, that config update has fully resolved the issue! It has also resolved the issue with v7.1.3.
